src/OrchardCore.Modules/OrchardCore.Contents/Filters/BuildDisplayFilter.cs

```diff
@@ -13,9 +14,19 @@ public class BuildDisplayFilter : ILiquidFilter
     {
         public async Task<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, TemplateContext ctx)
         {
-            var contentItem = input.ToObjectValue() as ContentItem;
+            var obj = input.ToObjectValue();
 
-            if (contentItem == null)
+            if (!(obj is ContentItem contentItem))
+            {
+                contentItem = null;
+
+                if (obj is JObject jObject)
+                {
+                    contentItem = jObject.ToObject<ContentItem>();
+                }
+            }
+
+            if (contentItem == null || contentItem.ContentItemId == null)
```

Comment:
`if (contentItem?.ContentItemId == null)`

Also please add a comment for why you test `null` here, I will forget ;)

Reply:
Okay